server/src/com/cloud/template/TemplateAdapterBase.java

```diff
@@ -171,14 +171,14 @@ public TemplateProfile prepare(boolean isIso, long userId, String name, String d
             sshkeyEnabled = Boolean.FALSE;
         }
 
-        boolean isAdmin = _accountMgr.isRootAdmin(templateOwner.getId());
+        boolean isRootAdmin = _accountMgr.isRootAdmin(templateOwner.getId());
         boolean isRegionStore = false;
         List<ImageStoreVO> stores = _imgStoreDao.findRegionImageStores();
         if (stores != null && stores.size() > 0) {
             isRegionStore = true;
         }
 
-        if (!isAdmin && zoneIdList == null && !isRegionStore ) {
+        if (!isRootAdmin && zoneIdList == null && !isRegionStore ) {
             // domain admin and user should also be able to register template on a region store
```

Comment:
Let's update this comment:

// Root Admin and user should also be able to register template on a region store

Reply:
Done